pandas/tests/frame/test_apply.py

```diff
@@ -116,6 +116,15 @@ def test_apply_with_reduce_empty(self):
         # Ensure that x.append hasn't been called
         assert x == []
 
+    @pytest.mark.parametrize("func", ["sum", "prod", "any", "all"])
+    def test_apply_funcs_over_empty(self, func):
+        # GH 28213
+        df = DataFrame(columns=["a", "b", "c"])
+
+        result = df.apply(eval(func))
```

Comment:
Instead of using `eval` can you just specify `@pytest.mark.parametrize("func", [np.sum, np.prod, np.any, np.all])`

Reply:
Is there a way to get that to work with `operator.methodcaller`?  It seems to want a string as the method name.

Reply:
You can just use `getattr`:

```
result = df.apply(getattr(np, 'func'))
expected = getattr(df, 'func')()
```